src/DataObject/GridColumnConfig/Operator/Substring.php

```diff
@@ -84,13 +91,35 @@ public function getLabeledValue(array|ElementInterface $element): ResultContaine
             if ($isArrayType) {
                 $result->value = $valueArray;
             } else {
-                $result->value = $valueArray[0];
+                $result->value = $valueArray[0] ?? '';
             }
         }
 
         return $result;
     }
 
+    private function convertToString(mixed $value): string
+    {
+        if (is_array($value)) {
+            $output = implode(
+                ' ',
+                array_map(
+                    fn($v) => is_scalar($v) ? (string)$v : '',
+                    array_values($value)
+                )
+            );
+        } elseif (is_object($value)) {
+            $output = method_exists($value, '__toString')
```

Comment:
```suggestion
            $output = $value instanceof \Stringable
```

(every object that implements this method automatically also implements this interface, even if not explicitly declared)

Reply:
ohhh.. i didn't know that or i just forgot about this magic , thank you for the insight
it's exactly what i needed, as `method_exists` is not really optimal other than for BC/FC and compatibility stuff 😄